server/keyspace/gc.go

```diff
@@ -0,0 +1,186 @@
+// Copyright 2023 TiKV Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package keyspace
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/pingcap/failpoint"
+	"github.com/pingcap/kvproto/pkg/keyspacepb"
+	"github.com/pingcap/log"
+	"github.com/tikv/pd/pkg/member"
+	"github.com/tikv/pd/server/config"
+	"go.uber.org/zap"
+)
+
+// Some common per-loop config.
+const (
+	// gcPerLoopTimeout specifies timeout for a single loop.
+	gcPerLoopTimeout = 30 * time.Second
```

Comment:
Generally, cleaning up `placement rule/region label/etcd path` is fast, but cleaning up `data range` is slower. 
I'm not sure if `30s` is a bit short. After all, there will be multiple keyspaces during a cleanup cycle.